src/Peering/Peering/PeeringService/Prefix/DeleteAzurePeeringServicePrefixCommand.cs

```diff
@@ -0,0 +1,165 @@
+        public string PeeringServiceName { get; set; }
+
+        /// <summary>
+        /// The resource  id
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceIdHelp,
+            ParameterSetName = Constants.ParameterSetNameByResourceId)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = Constants.ForceHelp)]
+        public SwitchParameter Force { get; set; }
+
+        /// <summary>
+        ///     The AsJob parameter to run in the background.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = Constants.AsJobHelp)]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        /// <summary>
+        /// The inherited Execute function.
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                this.ConfirmAction(
+                    Force.IsPresent,
+                    string.Format(Resources.ProcessMessage, this.Name),
+                    Resources.ContinueMessage,
+                    this.Name,
+                    () =>
+                    {
+                        if (this.ResourceId != null)
+                        {
+                            var resourceId = new ResourceIdentifier(this.ResourceId);
+                            this.ResourceGroupName = resourceId.ResourceGroupName;
+                            this.Name = resourceId.ResourceName;
+                            this.PeeringServiceName = resourceId.ParentResource.Split('/')?[1];
+                        }
+                        if (this.InputObject != null)
+                        {
+                            var resourceId = new ResourceIdentifier(this.InputObject.Id);
+                            this.ResourceGroupName = resourceId.ResourceGroupName;
+                            this.Name = resourceId.ResourceName;
+                            this.PeeringServiceName = resourceId.ParentResource.Split('/')?[1];
+                        }
+                        try
+                        {
+                            if (this.ShouldProcess(string.Format(Resources.ShouldProcessMessage, "null. Unless the -PassThru is present which it will return a boolean, true for success of false for failure.")))
+                            {
+                                this.PeeringServicePrefixesClient.Delete(this.ResourceGroupName, this.PeeringServiceName, this.Name);
+                                if (this.PassThru.IsPresent)
+                                {
+                                    WriteObject(true);
+                                }
+                            }
+                        }
+                        catch (Exception ex)
```

Comment:
@rileymckenna I don't think we want to catch this exception, I think we want to throw any exception that's received from the server